### PR TITLE
Refactor Stacktrace to not require logger

### PR DESCRIPTION
**Pull request by kstenerud**
## Goal

The new ANR native trace functionality will require native code to produce `Stacktrace` objects, however the constructors for `Stacktrace` require a logger, which is only used when converting JVM stack traces. Rather than passing a logger through native code, it's cleaner to move the "Java trace to Bugsnag trace format" code into a helper function.

## Design

This change moves the "java to bugsnag" constructor into a helper function in the object's companion. The `Stacktrace` object itself is now a POJO, containing data only.

## Testing

- Ran unit tests
- Ran example app, triggering crashes and verifying that the stack traces are correct in Java and native crashes.

### CODE REVIEW

File: bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt
Comment (fractalwrench) [inline bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt:9]: This should remain `internal` - the default visibility in Kotlin is public, so removing this modifier would expose `Stacktrace` as part of the public API.

File: bugsnag-android-core/src/test/java/com/bugsnag/android/StacktraceSerializationTest.kt
Comment (fractalwrench) [inline bugsnag-android-core/src/test/java/com/bugsnag/android/StacktraceSerializationTest.kt:21]: style nit: the indentation is inconsistent with other parameters

File: bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt
Comment (fractalwrench) [inline bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt:35]: The NDK will also need to limit the number of stackframes to 200 - this seems fine as long as the requirement isn't forgotten when implementing a similar method to achieve this via the JNI call.

**Review by fractalwrench — COMMENTED** [review]
The general approach seems ok here. I added a few minor comments inline, and there also seem to be a couple of compilation errors in the tests + static analysis warnings that will need resolving before this can be merged.

File: bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt
Comment (kstenerud) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt:9]: Hmm I had to expose it in order for the JNI plugin to see it. Is there a way to keep it internal but allow the JNI plugin visibility?

File: bugsnag-android-core/src/test/java/com/bugsnag/android/StacktraceSerializationTest.kt
Comment (kstenerud) [reply bugsnag-android-core/src/test/java/com/bugsnag/android/StacktraceSerializationTest.kt:21]: For some reason Android Studio insists on this indentation, and backspacing just moves the whole thing to the previous line. I had to manually select 4 spaces and cut them to get back to the original indentation :/

File: bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt
Comment (kstenerud) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt:35]: Yes, it will use the same `BUGSNAG_FRAMES_MAX ` as https://github.com/bugsnag/bugsnag-android/blob/master/bugsnag-plugin-android-ndk/src/main/jni/event.h#L166

File: bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt
Comment (fractalwrench) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt:9]: Ah, I see. Kotlin decompiles down to Java bytecode and [mangles names](https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#visibility) for methods in internal classes I believe. You could attempt using [`@JvmName`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-name/) or even making the method private to avoid name mangling, which is what the `AnrPlugin` does: https://github.com/bugsnag/bugsnag-android/blob/69aba9a05e5de94101a03ebb24bf49d7d6438085/bugsnag-plugin-android-anr/src/main/jni/anr_handler.c#L43

**Review by fractalwrench — APPROVED** [review]
LGTM pending CI

**Pull request merged**
